driver/src/main/java/org/neo4j/driver/v1/Path.java

```diff
@@ -61,7 +61,7 @@
     Node end();
 
     /** @return the number of segments in this path, which will be the same as the number of relationships */
-    long length();
+    int length();
```

Comment:
Hmm, true that the technical aspects of this do become tricky, since all the array types are `int` in Java. Yeah, fine, this will be easier for users - and if they are dealing with 2B+ long paths we'll likely need specialized path cursor types anyway.